### PR TITLE
Changed the docker and k8 config providers to return logs config as well

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Updated the utils method to extract config from docker labels and kubernetes annotations to return logs config.

### Motivation

Share a common logic to extract configs from docker labels and k8 annotations.

### CODE REVIEW

File: pkg/autodiscovery/providers/utils.go [hunk 1]
```diff
@@ -100,6 +101,25 @@ func buildTemplates(key string, checkNames []string, initConfigs, instances []in
...
+
+	checksConfigs, err := extractCheckTemplatesFromMap(key, input, prefix)
+	if err != nil {
+		return []integration.Config{}, err
+	}
+	configs = append(configs, checksConfigs...)
+
+	logsConfigs, err := extractLogsTemplatesFromMap(key, input, prefix)
+	if err != nil {
+		return []integration.Config{}, err
```
Comment (mfpierre) [inline pkg/autodiscovery/providers/utils.go]: I think you should return `configs` there to at least have the check configs if log fails

File: pkg/autodiscovery/providers/utils.go [hunk 2]
```diff
@@ -129,3 +149,16 @@ func extractTemplatesFromMap(key string, input map[string]string, prefix string)
...
+func extractLogsTemplatesFromMap(key string, input map[string]string, prefix string) ([]integration.Config, error) {
+	value, found := input[prefix+logsConfigPath]
+	if !found {
+		return []integration.Config{}, nil
+	}
+	logsConfigs, err := parseJSONValue(value)
+	if err != nil || len(logsConfigs) < 1 {
+		return []integration.Config{}, fmt.Errorf("in %s: %s", logsConfigPath, err)
+	}
+	return []integration.Config{{LogsConfig: logsConfigs[0], ADIdentifiers: []string{key}}}, nil
```
[the remark this answers is not in the document]
Comment (ajacquemot) [reply pkg/autodiscovery/providers/utils.go]: Because it's much more handy to use IMO.

**Review by mfpierre — APPROVED** [review]
LGTM

**Review by olivielpeau — APPROVED** [review]
Not very familiar with this part of the code, but LGTM 👍

**Pull request merged**